packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -3295,7 +3304,15 @@ function doubleInvokeEffectsInDEVIfNecessary(
     if (isInStrictMode && fiber.flags & Visibility) {
       // Double invoke effects on Offscreen's subtree only
       // if it is visible and its visibility has changed.
-      doubleInvokeEffectsOnFiber(root, fiber);
+
+      // For backwards compatibility, we don't unmount passive effects when a tree suspends.
+      // StrictMode needs to align with this.
+      const includePassiveEffects: boolean =
+        (fiber.stateNode._visibility & OffscreenPassiveEffectsConnected) !==
```

Comment:
Why is this line necessary? Seems like the tag check below should be sufficient. Regression test passes, too.

Reply:
Let me remove it. I added it because that's what the actual implementation does.